src/Flow.cpp

```diff
@@ -8733,6 +8740,27 @@ bool Flow::checkS1ConnState() {
 				   !(isTCPFlagSet(src2dst_tcp_flags,TH_RST)) && !(isTCPFlagSet(dst2src_tcp_flags,TH_RST))     /* NO RST */
 				   ));
 }
+/* **************************************************** */
+
+void Flow::updateSynFloodAlerts(bool connection_opened){
+    if((current_c_state != S0 && !connection_opened) || 
+    (current_c_state !=MINOR_NO_STATE && connection_opened))
+      return;
```

Comment:
If handled correctly in the calculateConnectionState() function, i don't think this check is needed

Reply:
These checks are mainly for two cases:

1.  Every data packet sent after the handshake trigger the checkS1ConnState(),
leading to unnecessarily decrementing the count.
2. If SYN packets are being resent from a client we will increment the count unnecessarily.

This logic could be in the calculateConnectionState() function but I thought it would be cleaner here.